### PR TITLE
Modify playbooks to use oc_obj module

### PR DESCRIPTION
**Pull request by mtnbikenc**

### CODE REVIEW

**Comment by mtnbikenc** [comment]
aos-ci-test

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 1]
```diff
@@ -1,12 +1,15 @@
 ---
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get node -o name
-  register: fluentd_hosts
+- name: Retrieve list of fluentd hosts
+  oc_obj:
+    state: list
+    kind: node
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/start_cluster.yaml]: No need to use the copy of the admin.kubeconfig here.

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 2]
```diff
@@ -19,107 +22,125 @@
...
 
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get dc -l component=es -o name -n {{openshift_logging_namespace}}
+- name: Retrieve elasticsearch
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=es"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/start_cluster.yaml]: No need to specify the kubeconfig here.

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 3]
```diff
@@ -19,107 +22,125 @@
...
-  check_mode: no
-  changed_when: no
 
 - name: start elasticsearch
   oc_scale:
     kind: dc
-    name: "{{object.split('/')[1]}}"
+    name: "{{ object }}"
     namespace: "{{openshift_logging_namespace}}"
     kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/start_cluster.yaml]: Not part of this PR, but kubeconfig doesn't need to be specified here either.

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 4]
```diff
@@ -19,107 +22,125 @@
...
 
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get dc -l component=kibana -o name -n {{openshift_logging_namespace}}
+- name: Retrieve kibana
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=kibana"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/start_cluster.yaml]: kubeconfig not needed

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 5]
```diff
@@ -19,107 +22,125 @@
...
-  check_mode: no
-  changed_when: no
 
 - name: start kibana
   oc_scale:
     kind: dc
-    name: "{{object.split('/')[1]}}"
+    name: "{{ object }}"
     namespace: "{{openshift_logging_namespace}}"
     kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/start_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 6]
```diff
@@ -19,107 +22,125 @@
...
 
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get dc -l component=curator -o name -n {{openshift_logging_namespace}}
+- name: Retrieve curator
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=curator"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/start_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 7]
```diff
@@ -19,107 +22,125 @@
...
-  check_mode: no
-  changed_when: no
 
 - name: start curator
   oc_scale:
     kind: dc
-    name: "{{object.split('/')[1]}}"
+    name: "{{ object }}"
     namespace: "{{openshift_logging_namespace}}"
     kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/start_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 8]
```diff
@@ -19,107 +22,125 @@
...
 
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get dc -l component=es-ops -o name -n {{openshift_logging_namespace}}
+- name: Retrieve elasticsearch-ops
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=es-ops"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/start_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 9]
```diff
@@ -19,107 +22,125 @@
...
-  check_mode: no
-  changed_when: no
 
 - name: start elasticsearch-ops
   oc_scale:
     kind: dc
-    name: "{{object.split('/')[1]}}"
+    name: "{{ object }}"
     namespace: "{{openshift_logging_namespace}}"
     kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/start_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 10]
```diff
@@ -19,107 +22,125 @@
...
 
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get dc -l component=kibana-ops -o name -n {{openshift_logging_namespace}}
+- name: Retrieve kibana-ops
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=kibana-ops"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/start_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 11]
```diff
@@ -19,107 +22,125 @@
...
-  check_mode: no
-  changed_when: no
 
 - name: start kibana-ops
   oc_scale:
     kind: dc
-    name: "{{object.split('/')[1]}}"
+    name: "{{ object }}"
     namespace: "{{openshift_logging_namespace}}"
     kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/start_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 12]
```diff
@@ -19,107 +22,125 @@
...
 
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get dc -l component=curator-ops -o name -n {{openshift_logging_namespace}}
+- name: Retrieve curator
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=curator-ops"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/start_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 13]
```diff
@@ -19,107 +22,125 @@
...
-  check_mode: no
-  changed_when: no
 
 - name: start curator-ops
   oc_scale:
     kind: dc
-    name: "{{object.split('/')[1]}}"
+    name: "{{ object }}"
     namespace: "{{openshift_logging_namespace}}"
     kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/start_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/stop_cluster.yaml [hunk 14]
```diff
@@ -1,11 +1,15 @@
 ---
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get node -o name
-  register: fluentd_hosts
+- name: Retrieve list of fluentd hosts
+  oc_obj:
+    state: list
+    kind: node
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/stop_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/stop_cluster.yaml [hunk 15]
```diff
@@ -18,101 +22,125 @@
...
 
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get dc -l component=es -o name -n {{openshift_logging_namespace}}
+- name: Retrieve elasticsearch
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=es"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/stop_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/stop_cluster.yaml [hunk 16]
```diff
@@ -18,101 +22,125 @@
...
   register: es_dc
-  changed_when: no
 
 - name: stop elasticsearch
   oc_scale:
     kind: dc
-    name: "{{object.split('/')[1]}}"
+    name: "{{ object }}"
     namespace: "{{openshift_logging_namespace}}"
     kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/stop_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/stop_cluster.yaml [hunk 17]
```diff
@@ -18,101 +22,125 @@
...
 
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get dc -l component=kibana -o name -n {{openshift_logging_namespace}}
+- name: Retrieve kibana
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=kibana"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/stop_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/stop_cluster.yaml [hunk 18]
```diff
@@ -18,101 +22,125 @@
...
   register: kibana_dc
-  changed_when: no
 
 - name: stop kibana
   oc_scale:
     kind: dc
-    name: "{{object.split('/')[1]}}"
+    name: "{{ object }}"
     namespace: "{{openshift_logging_namespace}}"
     kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/stop_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/stop_cluster.yaml [hunk 19]
```diff
@@ -18,101 +22,125 @@
...
 
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get dc -l component=curator -o name -n {{openshift_logging_namespace}}
+- name: Retrieve curator
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=curator"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/stop_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/stop_cluster.yaml [hunk 20]
```diff
@@ -18,101 +22,125 @@
...
   register: curator_dc
-  changed_when: no
 
 - name: stop curator
   oc_scale:
     kind: dc
-    name: "{{object.split('/')[1]}}"
+    name: "{{ object }}"
     namespace: "{{openshift_logging_namespace}}"
     kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/stop_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/stop_cluster.yaml [hunk 21]
```diff
@@ -18,101 +22,125 @@
...
 
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get dc -l component=es-ops -o name -n {{openshift_logging_namespace}}
+- name: Retrieve elasticsearch-ops
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=es-ops"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/stop_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/stop_cluster.yaml [hunk 22]
```diff
@@ -18,101 +22,125 @@
...
   register: es_dc
-  changed_when: no
 
 - name: stop elasticsearch-ops
   oc_scale:
     kind: dc
-    name: "{{object.split('/')[1]}}"
+    name: "{{ object }}"
     namespace: "{{openshift_logging_namespace}}"
     kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/stop_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/stop_cluster.yaml [hunk 23]
```diff
@@ -18,101 +22,125 @@
...
 
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get dc -l component=kibana-ops -o name -n {{openshift_logging_namespace}}
+- name: Retrieve kibana-ops
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=kibana-ops"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/stop_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/stop_cluster.yaml [hunk 24]
```diff
@@ -18,101 +22,125 @@
...
   register: kibana_dc
-  changed_when: no
 
 - name: stop kibana-ops
   oc_scale:
     kind: dc
-    name: "{{object.split('/')[1]}}"
+    name: "{{ object }}"
     namespace: "{{openshift_logging_namespace}}"
     kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/stop_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/stop_cluster.yaml [hunk 25]
```diff
@@ -18,101 +22,125 @@
...
 
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get dc -l component=curator-ops -o name -n {{openshift_logging_namespace}}
+- name: Retrieve curator
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=curator-ops"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/stop_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/stop_cluster.yaml [hunk 26]
```diff
@@ -18,101 +22,125 @@
...
   register: curator_dc
-  changed_when: no
 
 - name: stop curator-ops
   oc_scale:
     kind: dc
-    name: "{{object.split('/')[1]}}"
+    name: "{{ object }}"
     namespace: "{{openshift_logging_namespace}}"
     kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/stop_cluster.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/upgrade_logging.yaml [hunk 27]
```diff
@@ -8,29 +8,36 @@
...
 # start ES so that we can run migrate script
-- command: >
-    {{openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get dc -l component=es -o name -n {{openshift_logging_namespace}}
+- name: Retrieve elasticsearch
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=es"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/upgrade_logging.yaml]: kubeconfig, again.

File: roles/openshift_logging/tasks/upgrade_logging.yaml [hunk 28]
```diff
@@ -8,29 +8,36 @@
...
 
-- command: >
-    {{ openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get pods -n {{openshift_logging_namespace}} -l component=es -o jsonpath='{.items[?(@.status.phase == "Running")].metadata.name}'
+- name: Wait for pods to stop
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=es"
+    namespace: "{{openshift_logging_namespace}}"
+    kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
```
Comment (detiber) [inline roles/openshift_logging/tasks/upgrade_logging.yaml]: kubeconfig, again.

File: playbooks/common/openshift-cluster/upgrades/initialize_nodes_to_upgrade.yml [hunk 29]
```diff
@@ -26,7 +23,9 @@
       ansible_ssh_user: "{{ g_ssh_user | default(omit) }}"
       ansible_become: "{{ g_sudo | default(omit) }}"
     with_items: " {{ groups['oo_nodes_to_config'] }}"
-    when: openshift_upgrade_nodes_label is defined and hostvars[item].openshift.common.hostname in nodes_to_upgrade
+    when:
+    - openshift_upgrade_nodes_label is defined
+    - hostvars[item].openshift.common.hostname in nodes_to_upgrade.results.results[0]['items'] | map(attribute='metadata') | map(attribute='name') | list
```
Comment (ingvagabund) [inline playbooks/common/openshift-cluster/upgrades/initialize_nodes_to_upgrade.yml]: This does not look nice but I guess there is no other way how to do it.

File: roles/openshift_logging/tasks/start_cluster.yaml [hunk 30]
```diff
@@ -19,107 +22,125 @@
...
 - name: start elasticsearch
   oc_scale:
     kind: dc
-    name: "{{object.split('/')[1]}}"
+    name: "{{ object }}"
     namespace: "{{openshift_logging_namespace}}"
     kubeconfig: "{{mktemp.stdout}}/admin.kubeconfig"
     replicas: 1
-  with_items: "{{es_dc.stdout_lines}}"
+  with_items: "{{ es_dc.results.results[0]['items'] | map(attribute='metadata') | map(attribute='name') | list }}"
```
Comment (ingvagabund) [inline roles/openshift_logging/tasks/start_cluster.yaml]: I wish there was a better way to do it, e.g. ``results.results[0].items | map(attribute='metadata.name')``

**Comment by mtnbikenc** [comment]
aos-ci-test

File: playbooks/common/openshift-cluster/upgrades/initialize_nodes_to_upgrade.yml [hunk 31]
```diff
@@ -26,7 +23,9 @@
       ansible_ssh_user: "{{ g_ssh_user | default(omit) }}"
       ansible_become: "{{ g_sudo | default(omit) }}"
     with_items: " {{ groups['oo_nodes_to_config'] }}"
-    when: openshift_upgrade_nodes_label is defined and hostvars[item].openshift.common.hostname in nodes_to_upgrade
+    when:
+    - openshift_upgrade_nodes_label is defined
+    - hostvars[item].openshift.common.hostname in nodes_to_upgrade.results.results[0]['items'] | map(attribute='metadata.name') | list
```
Comment (ashcrow) [inline playbooks/common/openshift-cluster/upgrades/initialize_nodes_to_upgrade.yml]: :+1:

File: playbooks/common/openshift-cluster/upgrades/post_control_plane.yml [hunk 32]
```diff
@@ -9,77 +9,94 @@
...
+      name: docker-registry
     register: _default_registry
-    failed_when: false
-    changed_when: false
     when: temp_skip_router_registry_upgrade is not defined
 
   - name: Update registry image to current version
-    when: _default_registry.rc == 0 and temp_skip_router_registry_upgrade is not defined
+    when:
+    - _default_registry.rc == 0
```
Comment (ashcrow) [inline playbooks/common/openshift-cluster/upgrades/post_control_plane.yml]: Double checking ... does the ``obj`` module pass the ``rc`` through? I see direct references to ``returncode`` but not ``rc``.

File: roles/openshift_logging/tasks/upgrade_logging.yaml [hunk 33]
```diff
@@ -8,29 +8,34 @@
...
-    {{ openshift.common.client_binary}} --config={{mktemp.stdout}}/admin.kubeconfig get pods -n {{openshift_logging_namespace}} -l component=es -o jsonpath='{.items[?(@.status.phase == "Running")].metadata.name}'
+- name: Wait for pods to stop
+  oc_obj:
+    state: list
+    kind: dc
+    selector: "component=es"
+    namespace: "{{openshift_logging_namespace}}"
   register: running_pod
-  until: running_pod.stdout != ''
+  until: running_pod.results.results.items[?(@.status.phase == "Running")].metadata.name != ''
```
Comment (ashcrow) [inline roles/openshift_logging/tasks/upgrade_logging.yaml]: Not sure about if this will need quoting or not.

File: playbooks/common/openshift-cluster/upgrades/post_control_plane.yml (hunk 32 above)
Comment (mtnbikenc) [inline playbooks/common/openshift-cluster/upgrades/post_control_plane.yml]: Good catch.  I'll fix this in another PR.
